src/components/shapesAndOntologies/ShapesAndOntologies.tsx

```diff
@@ -0,0 +1,101 @@
+import styles from "./ShapesAndOntologies.module.css";
+import Title from "../Title/Title";
+import { useTranslation } from "react-i18next";
+import { useContext, useEffect, useState } from "react";
+import { AuthContext } from "../../context/AuthContextProvider";
+import car from "../../assets/car.gif";
+import { ApiService } from "../../services/ApiService";
+import { mapOntologies, mapShapesAndOntologies, Ontology } from "../../utils/dataMapper";
+import SelfDescriptionCard from "../cards/SelfDescriptionCard";
+import { AuthContextValues } from "../../context/AuthContextValues";
+import Text from "../Text/Text";
+import SearchBar from "../searchBar/SearchBar";
+
+const ShapesAndOntologies = () => {
+    const { t } = useTranslation();
+    const authContext = useContext(AuthContext);
+    const [isLoading, setIsLoading] = useState(false);
+    const [originalOntologies, setOriginalOntologies] = useState<Ontology[]>([]);
+    const [filteredOntologies, setFilteredOntologies] = useState<Ontology[]>([]);
+
+    // todo im not completely satisfied with the way i fetch the ontologies. Would it be better to have the fetching logic in a separate file?
+    const fetchOntologies = async (authContext: AuthContextValues) => {
```

Comment:
Yes, let's extract this logic in a custom hook, then import that and use the referenced functions.

Reply:
I extracted the logic to datamapper